php/media/class-filter.php

```diff
@@ -697,6 +697,27 @@ public function catch_media_templates_maybe() {
 		}
 	}
 
+	/**
+	 * Fallback for render_block_data filter.
+	 *
+	 * @param string $block_content The block content about to be appended.
+	 * @param array  $block         The full block, including name and attributes.
+	 *
+	 * @return string
+	 */
+	public function filter_image_block_render_block( $block_content, array $block ) {
+		if ( 'core/image' === $block['blockName'] ) {
+			remove_filter( 'render_block', array( $this, 'filter_image_block_render_block' ), 10, 2 );
+
+			$filtered_block = $this->filter_image_block_pre_render( $block, $block );
+			$block_content = render_block( $filtered_block );
+
+			add_filter( 'render_block', array( $this, 'filter_image_block_render_block' ), 10, 2 );
+		}
+
+		return $block_content;
```

Comment:
@pereirinha  Instead of adding and removing the filter, you can add a value to the array to looks for and ignore if present. like below:
```suggestion
	public function filter_image_block_render_block( $block_content, array $block ) {
		if ( 'core/image' === $block['blockName'] && empty( $block['cld_render'] ) ) {
			$filtered_block                        = $this->filter_image_block_pre_render( $block, $block );
			$filtered_block['cld_render'] = true;
			$block_content                       = render_block( $filtered_block );
		}

		return $block_content;
```